### DISCUSSION TITLE
스터디 출석체크 앱 사용 방법

### DISCUSSION THREAD

**Discussion by justinyoo** [fusiondevkr/pl900#18] [0 공지사항]
우리 스터디 완주 조건 중 하나가 매 시간 출석하는 거, 다들 알고 계시죠?

그래서, 여러분 편하게 출석체크 하시라고 파워 앱 하나 만들었습니다. 파워 앱 사용 순서는 아래와 같습니다.

1. 기존 로그인 계정과 혼선을 방지하기 위해 프라이빗 브라우저를 엽니다.
    * 크롬의 incognito 모드, 엣지의 InPrivate 모드를 사용하시면 됩니다.
2. 등록하신 M365 개발자 계정으로 로그인합니다.
    * 명심하세요, 지난번 숙제 드렸던 M365 개발자 계정 만들고 등록하기 있죠? 그 계정으로 로그인하세요.
3. [출석체크(https://aka.ms/fdk/pl900/checkin)](https://aka.ms/fdk/pl900/checkin) 링크로 접속합니다.
4. 상단 드롭다운 리스트에서 본인 컴퓨터 혹은 모바일에 연결된 카메라를 선택합니다.
5. 최초 출석 체크시 본인 인증을 위해 사진을 6-7회 정도 찍습니다. `얼굴 인증` 버튼을 눌러주세요!
    * 최초 출첵 때만 6-7회 정도 셀카를 찍습니다. 두번째 부터는 한번만 찍으면 됩니다.
    * 인식률을 높이기 위해 본인 얼굴만 나오게끔 셀카를 찍어주세요!
    * 인식률이 85% 이상 넘어가면 본인 인증에 성공한 것입니다.
6. 이제 `출석 체크` 버튼을 눌러 주세요.
    * 출석 체크가 가능한 시간은 스터디 당일 20시-24시 사이입니다. 그 이외 시간대에는 출첵 하셔도 실패합니다.

<img src="https://user-images.githubusercontent.com/1538528/126118518-264aa365-a990-4e83-9c07-8067223d46a8.png" width="300" alt="인증 성공"/> <img src="https://user-images.githubusercontent.com/1538528/126118059-8d581c40-7758-4eae-a4c0-8ad8a527ab4e.png" width="300" alt="체크인 실패"/>

**Comment by jake-jeon9** [comment]
CheckInFlow.Run 실패: { "error": { "code": 502, "source": "asia-001.azure-apim.net", "clientRequestId": "8ab422b5-ac79-4509-8b33-b087e12e096a", "message": "BadGateway", "innerError": { "error": { "code": "NoResponse", "message": "서버가 업스트림 서버에서 응답을 받지 않았습니다. 요청 추적 ID는 '08585749137081626804042734315CU02'입니다." } } } }
오류로 출석체크가 안됩니다..

**Reply by Taevely** [reply]
출첵 오류가 나요. 30분 시도해봤는데 답을 찾지 못했습니다. ㅜ_ㅜ

**Reply by justinyoo** [reply]
커넥터에 문제가 생겼습니다. ㅠ 조치하겠습니다.

**Reply by justinyoo** [reply]
해결했습니다! 다음주에는 잘 되겠죠 ㄷㄷㄷ

![image](https://user-images.githubusercontent.com/1538528/126189995-216609a8-a4b7-4600-8c83-69d2c02117ba.png)

**Comment by danielkim360** [comment]
출첵합니다. (오류가 나서...)

**Reply by snowpeak1004** [reply]
출첵 오류가 나네요.. ㅠㅠ

**Reply by justinyoo** [reply]
해결했습니다! 다음주에는 잘 되겠죠 ㄷㄷㄷ

**Comment by yjchoi07** [comment]
출쳌 합니다! (오류)

**Reply by justinyoo** [reply]
해결했습니다! 다음주에는 잘 되겠죠 ㄷㄷㄷ

**Comment by coogie02** [comment]
출석체크합니다. 
아래와 같이 오류가 납니다... 
coogie02@hwolrd365.microsoft.com

CheckInFlow.Run 실패: { "error": { "code": 502, "source": "asia-001.azure-apim.net", "clientRequestId": "696d6897-71ca-4305-b1ef-324affa5e0ed", "message": "BadGateway", "innerError": { "error": { "code": "NoResponse", "message": "서버가 업스트림 서버에서 응답을 받지 않았습니다. 요청 추적 ID는 '08585749086238322231690202561CU08'입니다." } } } }

일찍 일어나야 되서 아래정보 남깁니다.. ㅠㅠ
![image](https://user-images.githubusercontent.com/87651960/126162955-59cdc75d-73c9-4693-b8dc-58defe7a05ec.png)

**Reply by justinyoo** [reply]
해결했습니다! 다음주에는 잘 되겠죠 ㄷㄷㄷ

**Comment by goodteacher1** [comment]
출첵합니다!(오류)

FaceRecogniser.FacesIdentify 실패: {"message":"Too many faces or no face detected","confidence":0.0,"isIdentified":false,"timestamp":"2021-07-19T12:53:47.3198109+00:00"}

CheckInFlow.Run 실패: { "error": { "code": 502, "source": "asia-001.azure-apim.net", "clientRequestId": "2b3977e6-9057-4ec4-b093-b04d33962606", "message": "BadGateway", "innerError": { "error": { "code": "NoResponse", "message": "서버가 업스트림 서버에서 응답을 받지 않았습니다. 요청 추적 ID는 '08585749076114591373157111340CU08'입니다." } } } }

**Reply by justinyoo** [reply]
해결했습니다! 다음주에는 잘 되겠죠 ㄷㄷㄷ

**Comment by yjchoi07** [comment]
오늘 출첵을 시도 하였습니다. 휴대폰 앱과 크롬 시크릿 창에서도 해보았는데 안되네요! 
대신 여기다가 출첵 하겠습니다!

**Reply by jake-jeon9** [reply]
저두.. 한 50번 넘게 다양한 방법으로 시도했으나 컨피던스 80미만으로만 평균 70 나와서.. 여기에 출첵 희망합니다.
로그 보시면 ㅠㅠ 다양한 방법을 해도 80을 못넘기네요..

**Reply by justinyoo** [reply]
@yjchoi07 안된다는 부분에 대한 자세한 설명이 필요합니다. 다른 분들이 됐는데 안됐다면 앱의 문제는 아닌 것 같아요.

**Reply by justinyoo** [reply]
@jake-jeon9 ㅠㅠ

**Comment by jake-jeon9** [comment]
![image](https://user-images.githubusercontent.com/71210250/126997840-273226be-899b-477e-9b42-c9e21d9e6610.png)

ㅜㅜ 혹시 제 학습된 개체 초기화 가능할가요 ㅠㅠ 이정도 안되는거면.... ㅜㅜ

**Reply by justinyoo** [reply]
@jake-jeon9 ㅠㅠ 사진을 일단 다 초기화 했습니다. 배경이 혹시나 너무 밝다든가, 막 후광이 비친다든가, 화질이 좀 열악하다든가 하면 인식률에 문제가 생길 가능성이 높습니다. 그리고, 가급적이면 배경이 단순하면 제일 좋아요. 다시 한 번 해 보시겠어요? 카메라 꼭 쳐다보시구요!

**Reply by jake-jeon9** [reply]
@justinyoo 네!! 지금 다시해서 95%나왔습니다 ㅎㅎ 감사합니다!

**Comment by zzizziv** [comment]
네가 이기냐 내가 이기냐 해봤지만
제가 졌습니다.
카메라가 문제인지...인식이 잘 안되네요.

**Reply by justinyoo** [reply]
@zzizziv ㅠㅠ 일단 사진 초기화 시켜볼께요. 배경이 얼굴보다 더 많으면 인식률이 떨어집니다.

**Comment by yjchoi07** [comment]
출첵

**Reply by justinyoo** [reply]
여기 말고 앱에서 하세요
